lib/core/src/Cardano/Wallet/Network.hs

```diff
@@ -276,7 +276,9 @@ follow
     -- ^ Getter on the abstract 'block' type
     -> IO ()
 follow nl tr cps yield rollback header =
-    sleep 0 (initCursor nl cps)
+    -- NOTE: We start with sleeping for delay0. Not perfect,
+    -- but shouldn't matter.  We cannot use 0, or the backoff will break.
+    sleep delay0 (initCursor nl cps)
```

Comment:
Hmm. This is quite unsatisfactory. What about fixing `retryDelay` instead and make sure it uses `delay0` instead of `2*delay` when `delay == 0` ?

Reply:
>  What about fixing retryDelay instead and make sure it uses delay0 instead of 2*delay when delay == 0

That's odd, but it works better. I wouldn't call it "fixing retryDelay"…

```haskell
    retryDelay :: Int -> Int
    retryDelay 0 = delay0
    retryDelay delay = min (2*delay) (60 * delay0)
```

But, actually I think, we _can_ just `step` instead of `sleep`, which would be what we want 🤔 … with some tweaking…

Reply:
`step` doesn't have proper error handling, that's why we `sleep` here :+1:

Reply:
> That's odd, but it works better.

What is odd exactly and what works better?

Reply:
It shouldn't sleep initially, so it works better.

I find it odd that we rely on our backoff function to _not sleep initially_. `0` becomes a magic value.

```
0
1
2
4
8
```

But sure. Maybe it's not that bad.